File: 2024/August/03/MakeTwoArraysEqualbyReversingSubarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        unordered_map<int,int> mp;
        for(int ele : target)
            mp[ele]++;
        for(int ele : arr)
            mp[ele]--;
        for(auto pairs : mp)
            if(pairs.second)
                return false;
        return true;
    }
};

File: 2024/August/03/MakeTwoArraysEqualbyReversingSubarrays.py
#! /usr/bin/env python3

class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        return  Counter(target)==Counter(arr)
        '''
        count = defaultdict(int)
        for ele1,ele2 in zip(target,arr):
            count[ele1]+=1
            count[ele2]-=1
        for cnt in count:
            if count[cnt]:
                return False
        return True
        '''    
